ledger/apply/application.go

```diff
@@ -220,21 +228,30 @@ func optInApplication(balances Balances, sender basics.Address, appIdx basics.Ap
 		return err
 	}
 
+	if !balances.ConsensusParams().Application {
+		return fmt.Errorf("cannot opt in app %d for %s: applications not supported", appIdx, sender.String())
+	}
+
```

Comment:
We seem awfully deep to be checking this. Surely we rejected the appl transaction if the current proto doesn't support it? (Also, no need to call .String())

Reply:
agree. I'll remove this check both from `createApplication` as well as from `optInApplication`.

Reply:
updated.